pkg/fargate/fargate.go

```diff
@@ -61,6 +62,6 @@ func toSelectors(in []ekstypes.FargateProfileSelector) []api.FargateProfileSelec
 // Unauthorized errors are of the form:
 //   AccessDeniedException: Account <account> is not authorized to use this service
 func IsUnauthorizedError(err error) bool {
-	var accessDeniedErr *ekstypes.AccessDeniedException
-	return errors.As(err, &accessDeniedErr)
+	var apiErr smithy.APIError
+	return errors.As(err, &apiErr) && apiErr.ErrorCode() == "AccessDeniedException"
```

Comment:
should we add a test for this?

Reply:
The error is constructed by the AWS SDK from the HTTP response, it'd require mimicking the SDK logic.